### ISSUE TITLE
Decorated function can not return right value. 

### ISSUE THREAD

**Issue by honggaruy** [allrod5/parameters-validation#2]
I think I found a bug.

**How to reproduce:**
```python
from parameters_validation import non_null, validate_parameters

@validate_parameters
def concat(front: str, back: non_null(str)):
    result = None
    if front:
        result = front + '-' + back
    print('inner_result = {}'.format(result))
    return result

result_str = concat('one' , 'two')
print('outer_result = {}'.format(result_str)
```
step 1: Run above coce
step 2: 'inner_result' should be same with 'outer_result'.  But they are different
step 3: If you comment out '@validate_parameters' line, the output has no problem.

**Comment by roo-oliv** [comment]
Hi @honggaruy, thanks for the report with a reproducible code.

I was able to reproduce the issue in `parameters-validation` version 1.1.0 using Python 3.6.8.

I'm currently investigating this.

**Comment by roo-oliv** [comment]
@honggaruy a fix for this bug was released in version 1.1.1 and is already available at PyPI. Thanks for your help on this bug!

**Comment by honggaruy** [comment]
Thanks for the new release! Now I can use this library again.
